core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -129,8 +129,8 @@
   @Override
   public void write(Iterator<Product2<K, V>> records) throws IOException {
     assert (partitionWriters == null);
-    ShuffleMapOutputWriter mapOutputWriter = shuffleExecutorComponents
-        .createMapOutputWriter(shuffleId, mapId, mapTaskAttemptId, numPartitions);
+    ShuffleMapOutputWriter mapOutputWriter = shuffleExecutorComponents.createMapOutputWriter(
+        shuffleId, -1, mapId, mapTaskAttemptId, numPartitions);
```

Comment:
Why are we always passing in -1 here?

Reply:
This PR created for quick reviewing of API changes, you can see the real scenario of shuffleGeneraionId here :)